### ISSUE TITLE
[Super AI] [Good First Issue] Convert LLaMa to XFormer Attention

### ISSUE THREAD

**Issue by new5558** [OpenThaiGPT/openthaigpt-pretraining#72]
Description:
Although the Official Implementation of [LLaMa Model](https://github.com/facebookresearch/llama) mentioned they used [XFormer Attention](https://github.com/facebookresearch/xformers) in their paper to have efficient memory consumption when training, did not provide an example of doing so in their [inference code](https://github.com/facebookresearch/llama/blob/main/llama/model.py). In this task, we want to convert the original code from LLaMA to support Xformer Attetnion by doing

1. Download the model https://github.com/facebookresearch/llama/pull/73 and try to run inference using their official implementation code. (6.7B Version is enough)
2. Convert to both:
- Pytorch 2.0 Attention: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
- Xformer Attention 
3. Try to run inference again and get the same result (logits)

Deliverables:
- Convert/Inference Code Script
- PR
- Unit Test to make sure converted model have similar output to the original one
